### PR TITLE
[complex32, jiterator] cos, sinh, cosh, tanh

### PR DESCRIPTION
**Pull request by khushi-411**
Follows: #74537 and #74748

cc @kshitij12345!

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -14914,10 +14914,7 @@ def error_inputs_mean(op_info, device, **kwargs):
                    ref=np.sin,
                    dtypes=all_types_and_complex_and(torch.bool, torch.bfloat16),
                    dtypesIfCUDA=all_types_and_complex_and(torch.chalf, torch.bool, torch.half, torch.bfloat16),
-                   # TODO: Add torch.chalf backward dtype support. Currently, we get:
-                   # AssertionError: The supported dtypes for sin on device type cuda are incorrect!
-                   # The following dtypes did not work in backward but are listed by the OpInfo: {torch.complex32}.
-                   backward_dtypesIfCUDA=floating_and_complex_types_and(torch.half, torch.bfloat16),
+                   backward_dtypesIfCUDA=floating_and_complex_types_and(torch.chalf, torch.half, torch.bfloat16),
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:14917]: You can remove this since it is all the valid backward dtypes now.

**Review by kshitij12345 — APPROVED** [review]
Except for one minor change

LGTM! Thanks @khushi-411

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (khushi-411) [reply torch/testing/_internal/common_methods_invocations.py:14917]: Removed that. Thanks, @kshitij12345! :)

**Comment by khushi-411** [comment]
Hi,
Gentle ping, @anjali411! :)

**Review by anjali411 — APPROVED** [review]
LGTM! Thanks @khushi-411

**Pull request closed without merging**